plugin/hack.vim

```diff
@@ -107,11 +107,34 @@ function! hack#toggle()
   endif
 endfunction
 
+function! hack#format(from, to)
+  if !executable(g:hack#hh_format)
+    echo 'g:hack#hh_format not executable'
+  endif
+
+  if &modified
+    echo 'Error[hack]: buffer has unsaved changes'
+    return
+  endif
+  let frombyte = line2byte(a:from)
+
+  let tobyte = line2byte(a:to) + strlen(getline(a:to))
+
+  execute a:from.','.a:to.' ! '.g:hack#hh_format.
+    \ ' --from '.frombyte.' --to '.tobyte.
+    \ ' --root '.g:hack#root.' '.expand('%:p')
+  let tmp = g:hack#enable
+  " Disable auto checking
+  let g:hack#enable = 0
+  silent write
+  let g:hack#enable = tmp
```

Comment:
well, I've tested the feature myself and I find it a lot more usable if it saves immediately - the buffer is already saved, and the change you apply is formatting only, so it's really convenient to have it saved without doing an (expensive and sometimes slow) hack check